yt/frontends/enzo/data_structures.py

```diff
@@ -370,20 +370,29 @@ def _detect_active_particle_fields(self):
                 self.dataset.particle_types = new_ptypes
                 self.dataset.particle_types_raw = new_ptypes
                 continue
-            gs = self.grids[select_grids > 0]
-            g = gs[0]
-            handle = h5py.File(g.filename, mode="r")
-            node = handle["/Grid%08i/Particles/" % g.id]
-            for ptype in (str(p) for p in node):
-                if ptype not in _fields:
-                    continue
-                for field in (str(f) for f in node[ptype]):
-                    _fields[ptype].append(field)
-                    sh = node[ptype][field].shape
-                    if len(sh) > 1:
-                        self.io._array_fields[field] = (sh[1],)
-                fields += [(ptype, field) for field in _fields.pop(ptype)]
-            handle.close()
+            if ptype != "DarkMatter":
+                gs = self.grids[select_grids > 0]
+                g = gs[0]
+                handle = h5py.File(g.filename, "r")
```

Comment:
I can't make this a suggestion, but `h5py.File` now implements the context manager protocol. In other words:
```python
handle = h5py.File(...)
...
handle.close()
```
can (and should) be refactored into
```python
with h5py.File(...) as handle:
    ...
```

Reply:
Interesting. So everything between the 'with' and the 'close is indented and in that scope? Is that how that works?

Reply:
that's it. Basically it works as a the builtin `open` function. At the cost of adding one indentation level to your code, you gain confidence in that the file will always be closed correctly, even if you forget to manually close it, or if an exception is raised before you do.

```python
with open(...) as handle:
    ...
```
is essentially the same as
```python
handle = open(...)
try:
    ...
finally:
    handle.close()
```